<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.2.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":true},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":"valine","storage":true,"lazyload":false,"nav":null,"activeClass":"valine"},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.json"};
  </script>

  <meta name="description" content="以下内容均为复制生信菜鸟团的内容，为miRNA-seq的分析教程。">
<meta property="og:type" content="article">
<meta property="og:title" content="miRNA-seq生信分析（转生信菜鸟团）">
<meta property="og:url" content="http://example.com/2020/12/11/2020-12-11-miRNA-seq/index.html">
<meta property="og:site_name" content="滴水穿石">
<meta property="og:description" content="以下内容均为复制生信菜鸟团的内容，为miRNA-seq的分析教程。">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="http://www.bio-info-trainee.com/wp-content/uploads/2016/07/miRNA-heatmap.png">
<meta property="article:published_time" content="2020-12-11T07:00:00.000Z">
<meta property="article:modified_time" content="2020-12-11T07:00:00.000Z">
<meta property="article:author" content="ShuilinL">
<meta property="article:tag" content="miRNA-seq">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://www.bio-info-trainee.com/wp-content/uploads/2016/07/miRNA-heatmap.png">

<link rel="canonical" href="http://example.com/2020/12/11/2020-12-11-miRNA-seq/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-CN'
  };
</script>

  <title>miRNA-seq生信分析（转生信菜鸟团） | 滴水穿石</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<link rel="alternate" href="/atom.xml" title="滴水穿石" type="application/atom+xml">
</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="切换导航栏">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">滴水穿石</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">种一棵树最好的时间是十年前，其次是现在</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>首页</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>关于</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>标签<span class="badge">28</span></a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>分类<span class="badge">17</span></a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>归档<span class="badge">17</span></a>

  </li>
        <li class="menu-item menu-item-guestbook">

    <a href="/guestbook/" rel="section"><i class="fa fa-book fa-fw"></i>留言板</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>搜索
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="搜索..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="http://example.com/2020/12/11/2020-12-11-miRNA-seq/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/headshot.jpg">
      <meta itemprop="name" content="ShuilinL">
      <meta itemprop="description" content="Bioinformatics">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="滴水穿石">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          miRNA-seq生信分析（转生信菜鸟团）
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">发表于</span>

              <time title="创建时间：2020-12-11 15:00:00" itemprop="dateCreated datePublished" datetime="2020-12-11T15:00:00+08:00">2020-12-11</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Pipeline/" itemprop="url" rel="index"><span itemprop="name">Pipeline</span></a>
                </span>
                  ，
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Pipeline/miRNA-seq/" itemprop="url" rel="index"><span itemprop="name">miRNA-seq</span></a>
                </span>
            </span>

          
            <span id="/2020/12/11/2020-12-11-miRNA-seq/" class="post-meta-item leancloud_visitors" data-flag-title="miRNA-seq生信分析（转生信菜鸟团）" title="阅读次数">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">阅读次数：</span>
              <span class="leancloud-visitors-count"></span>
            </span>
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Valine：</span>
    
    <a title="valine" href="/2020/12/11/2020-12-11-miRNA-seq/#valine-comments" itemprop="discussionUrl">
      <span class="post-comments-count valine-comment-count" data-xid="/2020/12/11/2020-12-11-miRNA-seq/" itemprop="commentCount"></span>
    </a>
  </span>
  
  

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>以下内容均为复制生信菜鸟团的内容，为miRNA-seq的分析教程。<br><a id="more"></a> </p>
<h1 id="构建miRNA-seq数据分析环境"><a href="#构建miRNA-seq数据分析环境" class="headerlink" title="构建miRNA-seq数据分析环境"></a>构建miRNA-seq数据分析环境</h1><p>最近在生信技能树分享了几个miRNA的靶向基因的查询工具，分别是：</p>
<ul>
<li><a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s/n_UncYeGIQFLneTMK2rTXQ">microRNAs靶基因数据库哪家强</a></li>
<li><a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s/dkVBcls3f0acXtPnFSXeWw">使用miRNAtap数据源提取miRNA的预测靶基因结果</a></li>
<li><a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s/13m_XmSuFgiLvt3EQhJVOQ">对miRNA进行go和kegg等功能数据库数据库注释</a></li>
</ul>
<p><strong>很多粉丝留言想听miRNA-seq数据分析流程</strong>，主要是上游分析流程，因为下游其实就是表达矩阵分析。跟RNA-seq拿到的counts矩阵是类似的分析策略，只不过是miRNA-seq热度已经过去了，我也仅仅是五年前接触过一次。</p>
<p>其实miRNA-seq数据上游分析有两个方案，一个是仅仅针对已知的miRNA进行定量，这样的话无需比对到物种参考基因组，仅仅是比对到miRNA序列合集即可。另外一个挖掘新的miRNA，主要是考虑到人类对miRNA的认知的不停的进步。当然，如果你想系统性学习，可以参考<strong>我五年前</strong>在生信菜鸟团<strong>自学miRNA-seq分析系列笔记</strong>：</p>
<ul>
<li>自学miRNA-seq分析第八讲~miRNA-mRNA表达相关下游分析 | 生信菜鸟团</li>
<li>自学miRNA-seq分析第七讲~miRNA样本配对mRNA表达量获取 | 生信菜鸟团</li>
<li>自学miRNA-seq分析第六讲~miRNA表达量差异分析 | 生信菜鸟团</li>
<li>自学miRNA-seq分析第五讲~miRNA表达量获取 | 生信菜鸟团</li>
<li>自学miRNA-seq分析第四讲~测序数据比对 | 生信菜鸟团</li>
<li>自学miRNA-seq分析第三讲~公共测序数据下载 | 生信菜鸟团</li>
<li>自学miRNA-seq分析第二讲~学习资料的搜集 | 生信菜鸟团</li>
<li>自学miRNA-seq分析第一讲~文献选择与解读 | 生信菜鸟团</li>
</ul>
<p>当然了，如果你是微信阅读，也可以点击：<a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s/USc_aUA_loFLLljXzbmpFg">一篇文章学会miRNA-seq分析</a></p>
<h3 id="首先下载MIRBASE的MIRNA参考序列文件"><a href="#首先下载MIRBASE的MIRNA参考序列文件" class="headerlink" title="首先下载MIRBASE的MIRNA参考序列文件"></a>首先下载MIRBASE的MIRNA参考序列文件</h3><p>miRBase是miRNA研究领域最权威的数据库，提供miRNAs序列以及注释查询、定位、发卡序列查询，以及提供命名服务。截止到现在（2020-04-28 ），miRBase 22.1 共收录了271个物种的总共38589 条miRNA信息。其中人类的共收录了1917条pre-miRNA（前体），以及2656条成熟的miRNAs。见：<a target="_blank" rel="noopener" href="http://www.mirbase.org/">http://www.mirbase.org/</a></p>
<p>前体miRNA和成熟的miRNA的区别，前体miRNA长度一般是<strong>70–120 碱基，一般是茎环结果，也就是发夹结构，所以叫做hairpin。成熟之后，一般</strong>22 个碱基。</p>
<p>在 <a target="_blank" rel="noopener" href="http://www.mirbase.org/">http://www.mirbase.org/</a> 对应的ftp网站下载如下所示文件：</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">mkdir -p ~/reference/miRNA</span><br><span class="line"><span class="built_in">cd</span> ~/reference/miRNA</span><br><span class="line">wget ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.gz <span class="comment">##　38589　reads</span></span><br><span class="line">wget ftp://mirbase.org/pub/mirbase/CURRENT/mature.fa.zip <span class="comment">## 48885 reads </span></span><br><span class="line">wget ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.zip</span><br><span class="line">wget ftp://mirbase.org/pub/mirbase/CURRENT/genomes/hsa.gff3</span><br><span class="line">unzip hairpin.fa.zip</span><br><span class="line">unzip mature.fa.zip</span><br><span class="line">grep sapiens mature.fa |wc -l 　<span class="comment"># 2656 </span></span><br><span class="line">grep sapiens hairpin.fa |wc <span class="comment"># 1917</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## Homo sapiens</span></span><br><span class="line">perl -alne <span class="string">&#x27;&#123;if(/^&gt;/)&#123;if(/Homo/)&#123;$tmp=1&#125;else&#123;$tmp=0&#125;&#125;;next if $tmp!=1;s/U/T/g if !/&gt;/;print &#125;&#x27;</span> hairpin.fa &gt; hairpin.human.fa</span><br><span class="line">perl -alne <span class="string">&#x27;&#123;if(/^&gt;/)&#123;if(/Homo/)&#123;$tmp=1&#125;else&#123;$tmp=0&#125;&#125;;next if $tmp!=1;s/U/T/g if !/&gt;/;print &#125;&#x27;</span> mature.fa &gt; mature.human.fa</span><br><span class="line"><span class="comment"># 这里值得一提的是miRBase数据库下载的序列，居然都是用U表示的，也就是说就是miRNA序列，而不是转录成该miRNA的基因序列，而我们测序的都是基因序列。</span></span><br></pre></td></tr></table></figure>
<p>最后得到的文件如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">5.9M Mar 12 2018 hairpin.fa</span><br><span class="line">1.5M Apr 29 15:09 hairpin.fa.gz</span><br><span class="line">1.5M Apr 29 15:11 hairpin.fa.zip</span><br><span class="line">263K Apr 29 15:13 hairpin.human.fa</span><br><span class="line">523K Apr 29 15:11 hsa.gff3</span><br><span class="line">3.7M Mar 12 2018 mature.fa</span><br><span class="line">786K Apr 29 15:09 mature.fa.zip</span><br><span class="line">196K Apr 29 15:13 mature.human.fa</span><br></pre></td></tr></table></figure>
<p>构建miRNA-seq数据分析环境只需要使用上面的的文件即可。</p>
<h3 id="然后使用CONDA配置好软件环境"><a href="#然后使用CONDA配置好软件环境" class="headerlink" title="然后使用CONDA配置好软件环境"></a>然后使用CONDA配置好软件环境</h3><p>仍然是参考我五年前整理的流程，使用bowtie软件和<strong>SHRiMP</strong>软件进行比对，然后<strong>fastx_toolkit 和fastqc</strong>进行质量控制。</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">conda create -n miRNA </span><br><span class="line">conda activate miRNA </span><br><span class="line">conda install -y -c bioconda hisat2 bowtie samtools fastp bowtie2 fastx_toolkit fastqc</span><br><span class="line"><span class="comment"># sra-toolkits,subreads 也可以一并下载 </span></span><br><span class="line"><span class="comment"># conda search fastx_toolkit</span></span><br><span class="line"><span class="comment"># 耗费约1.2G的磁盘空间</span></span><br></pre></td></tr></table></figure>
<p>因为<strong>SHRiMP</strong>软件太多年没有更新，所以放弃它，反正比对这个过程，有bowtie就ok了。</p>
<h3 id="针对MIRBASE数据库下载的序列构建BOWTIE索引"><a href="#针对MIRBASE数据库下载的序列构建BOWTIE索引" class="headerlink" title="针对MIRBASE数据库下载的序列构建BOWTIE索引"></a>针对MIRBASE数据库下载的序列构建BOWTIE索引</h3><p>需要注意的是<strong>bowtie和bowtie2</strong>不一样哦：</p>
<ul>
<li><a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/index.shtml">http://bowtie-bio.sourceforge.net/index.shtml</a></li>
<li><a target="_blank" rel="noopener" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a></li>
</ul>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">bowtie-build mature.human.fa hsa-mature-bowtie-index</span><br><span class="line">bowtie-build hairpin.human.fa hsa-hairpin-bowtie-index</span><br></pre></td></tr></table></figure>
<p>得到的文件如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">4.2M Apr 29 15:42 hsa-hairpin-bowtie-index.1.ebwt</span><br><span class="line"> 20K Apr 29 15:42 hsa-hairpin-bowtie-index.2.ebwt</span><br><span class="line"> 17K Apr 29 15:42 hsa-hairpin-bowtie-index.3.ebwt</span><br><span class="line"> 39K Apr 29 15:42 hsa-hairpin-bowtie-index.4.ebwt</span><br><span class="line">4.2M Apr 29 15:42 hsa-hairpin-bowtie-index.rev.1.ebwt</span><br><span class="line"> 20K Apr 29 15:42 hsa-hairpin-bowtie-index.rev.2.ebwt</span><br><span class="line">4.2M Apr 29 15:41 hsa-mature-bowtie-index.1.ebwt</span><br><span class="line">7.1K Apr 29 15:41 hsa-mature-bowtie-index.2.ebwt</span><br><span class="line"> 24K Apr 29 15:41 hsa-mature-bowtie-index.3.ebwt</span><br><span class="line"> 15K Apr 29 15:41 hsa-mature-bowtie-index.4.ebwt</span><br><span class="line">4.2M Apr 29 15:41 hsa-mature-bowtie-index.rev.1.ebwt</span><br><span class="line">7.1K Apr 29 15:41 hsa-mature-bowtie-index.rev.2.ebwt</span><br></pre></td></tr></table></figure>
<h3 id="下载测试数据"><a href="#下载测试数据" class="headerlink" title="下载测试数据"></a>下载测试数据</h3><p>这里我们仍然是使用 RNA expression profiling of human iPSC-derived cardiomyocytes in a cardiac hypertrophy model. <em>PLoS One</em> 2014;9(9):e108051. PMID: <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pubmed/25255322">25255322</a> 文章里面的 <a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60292">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60292</a> 数据集，如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">GSM1470353 control-CM, experiment1</span><br><span class="line">GSM1470354 ET1-CM, experiment1</span><br><span class="line">GSM1470355 control-CM, experiment2</span><br><span class="line">GSM1470356 ET1-CM, experiment2</span><br><span class="line">GSM1470357 control-CM, experiment3</span><br><span class="line">GSM1470358 ET1-CM, experiment3</span><br></pre></td></tr></table></figure>
<p>对应的</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">SRR1542714</span><br><span class="line">SRR1542715</span><br><span class="line">SRR1542716</span><br><span class="line">SRR1542717</span><br><span class="line">SRR1542718</span><br><span class="line">SRR1542719</span><br></pre></td></tr></table></figure>
<p>仍然是参考：<a target="_blank" rel="noopener" href="https://mp.weixin.qq.com/s/USc_aUA_loFLLljXzbmpFg">一篇文章学会miRNA-seq分析</a>，简单看看规律</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">ftp:&#x2F;&#x2F;ftp.sra.ebi.ac.uk&#x2F;vol1&#x2F;srr&#x2F;SRR154&#x2F;004&#x2F;SRR1542714； ftp:&#x2F;&#x2F;ftp.sra.ebi.ac.uk&#x2F;vol1&#x2F;fastq&#x2F;SRR154&#x2F;004&#x2F;SRR1542714&#x2F;SRR1542714.fastq.gz</span><br><span class="line"># 从14到19</span><br><span class="line"># 可以对这6个样本，分开prefetch</span><br><span class="line">~&#x2F;biosoft&#x2F;sratoolkit&#x2F;sratoolkit.2.8.2-1-centos_linux64&#x2F;bin&#x2F;prefetch SRR1542714</span><br></pre></td></tr></table></figure>
<p>脚本如下：</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">mkdir -p ~/reference/miRNA</span><br><span class="line"><span class="built_in">cd</span> ~/reference/miRNA </span><br><span class="line"><span class="comment"># step1 : download raw data</span></span><br><span class="line">mkdir miRNA_test &amp;&amp; <span class="built_in">cd</span> miRNA_test</span><br><span class="line"><span class="built_in">echo</span> &#123;14..19&#125; |sed <span class="string">&#x27;s/ /\n/g&#x27;</span> |<span class="keyword">while</span> <span class="built_in">read</span> id; \</span><br><span class="line"><span class="keyword">do</span> (~/biosoft/sratoolkit/sratoolkit.2.8.2-1-centos_linux64/bin/prefetch SRR15427<span class="variable">$&#123;id&#125;</span> ) ;\</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="comment"># 下面是另外一个课题，可以参考比较</span></span><br><span class="line"><span class="built_in">echo</span> &#123;44..59&#125; |sed <span class="string">&#x27;s/ /\n/g&#x27;</span> |<span class="keyword">while</span> <span class="built_in">read</span> id; \</span><br><span class="line"><span class="keyword">do</span> (~/biosoft/sratoolkit/sratoolkit.2.8.2-1-centos_linux64/bin/prefetch SRR77077<span class="variable">$&#123;id&#125;</span> ) ;\</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"><span class="comment"># 另外，这样的下载方式，在中国大陆会非常慢！！！</span></span><br><span class="line"><span class="comment"># 建议换成aspera哈</span></span><br></pre></td></tr></table></figure>
<p>得到的<strong>sra文件</strong>如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">33M Apr 29 16:07 SRR1542714.sra</span><br><span class="line">31M Apr 29 16:07 SRR1542715.sra</span><br><span class="line">50M Apr 29 16:07 SRR1542716.sra</span><br><span class="line">24M Apr 29 16:07 SRR1542717.sra</span><br><span class="line">52M Apr 29 16:07 SRR1542718.sra</span><br><span class="line">34M Apr 29 16:07 SRR1542719.sra</span><br></pre></td></tr></table></figure>
<p>然后批量<strong>转为fq文件</strong>， 代码如下：</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">dump=/home/yb77613/biosoft/sratoolkit/sratoolkit.2.8.2-1-centos_linux64/bin/fastq-dump</span><br><span class="line"><span class="built_in">echo</span> &#123;14..19&#125; |sed <span class="string">&#x27;s/ /\n/g&#x27;</span> |<span class="keyword">while</span> <span class="built_in">read</span> id; \</span><br><span class="line"><span class="keyword">do</span> ( <span class="variable">$dump</span> -O ./ --gzip --split-3 SRR15427<span class="variable">$&#123;id&#125;</span> ) ;\</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p>文件如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">50M Apr 29 16:14 SRR1542714.fastq.gz</span><br><span class="line">47M Apr 29 16:15 SRR1542715.fastq.gz</span><br><span class="line">75M Apr 29 16:15 SRR1542716.fastq.gz</span><br><span class="line">35M Apr 29 16:15 SRR1542717.fastq.gz</span><br><span class="line">81M Apr 29 16:16 SRR1542718.fastq.gz</span><br><span class="line">50M Apr 29 16:16 SRR1542719.fastq.gz</span><br></pre></td></tr></table></figure>
<h3 id="质控和清洗测序数据"><a href="#质控和清洗测序数据" class="headerlink" title="质控和清洗测序数据"></a>质控和清洗测序数据</h3><p>清洗前后，都走一下fastqc图表，清洗主要是fastx_toolkit修剪，代码如下：</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">ls *gz |<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$id</span> </span><br><span class="line"><span class="comment"># fastqc $id </span></span><br><span class="line">zcat <span class="variable">$id</span>| fastq_quality_filter -v -q 20 -p 80 -Q33 -i - -o tmp ;</span><br><span class="line">fastx_trimmer -v -f 1 -l 27 -m 15 -i tmp -Q33 -z -o <span class="variable">$&#123;id%%.*&#125;</span>_clean.fq.gz ;</span><br><span class="line"><span class="comment"># fastqc $&#123;id%%.*&#125;_clean.fq.gz </span></span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p>fastq_quality_filter和fastx_trimmer两个命令，都是来自于fastx_toolkit软件包：</p>
<ul>
<li>fastq_quality_filter 代表根据质量过滤序列</li>
<li>fastx_trimmer 代表缩短FASTQ或FASTQ文件中的读数，根据质量修剪（剪切）序列。</li>
</ul>
<p>其参数详解如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">#运行一次查看是否可以成功运行</span><br><span class="line">#fastq_quality_filter 代表根据质量过滤序列</span><br><span class="line">#-v,即verbose,报告序列的数目</span><br><span class="line">#-q,即quality,保留碱基所要求的最小质量评分，低于此值将会去除</span><br><span class="line">#-p,即percent,即序列中超过最小质量评分的碱基数目的最小百分率，低于这个比率将删除</span><br><span class="line">#-Q33,即phred33,默认是按照phred64作为参考的</span><br><span class="line">#-i,即infile,代表输入文件，注意不能是压缩文件，可以是FASTA&#x2F;FASTQ都行。默认是STDIN，即标准输入</span><br><span class="line">#-o,即outfile,代表输出文件（注意不是output dir即输出目录，此处输出是一个文件而不是文件夹），默认是STDOUT即标准输出，指定则输出到指定文件</span><br><span class="line">fastq_quality_filter -v -q 20 -p 80 -Q33 -i SRR1542714.1.fastq -o tmp #生成第一步处理的临时文件</span><br><span class="line"></span><br><span class="line"># 中间文件是 tmp ，到时候可以删除掉。</span><br><span class="line"></span><br><span class="line">#fastx_trimmer 代表缩短FASTQ或FASTQ文件中的读数，根据质量修剪（剪切）序列。</span><br><span class="line">#-v,即verbose,报告序列的数目</span><br><span class="line">#-f,即first,代表保留第几个碱基，默认是保留第一个</span><br><span class="line">#-l,即last,代表保留最后的碱基，默认是整个reads</span><br><span class="line">#-i,即infile,代表输入文件，注意不能是压缩文件，可以是FASTA&#x2F;FASTQ都行。默认是STDIN，即标准输入</span><br><span class="line">#-z,即compress,代表的是使用Gzip压缩输出</span><br><span class="line">#-o,即outfile,代表输出文件（注意不是output dir即输出目录，此处输出是一个文件而不是文件夹），默认是STDOUT即标准输出，指定则输出到指定文件</span><br><span class="line">fastx_trimmer -v -f 1 -l 27 -i tmp -Q33 -z -o SRR1542714.1_clean.fq.gz #生成最后的文件</span><br></pre></td></tr></table></figure>
<p>所以 fastq_quality_filter 和 fastx_trimmer命令是有替代品的，就是需要去自行搜索，如果你是要建立好用的miRNA-seq数据分析环境，就需要自己耗费大量时间在里面哦。</p>
<p>最后得到的干净的测序数据文件如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">39M Apr 29 16:28 SRR1542714_clean.fq.gz</span><br><span class="line">37M Apr 29 16:28 SRR1542715_clean.fq.gz</span><br><span class="line">49M Apr 29 16:29 SRR1542716_clean.fq.gz</span><br><span class="line">18M Apr 29 16:29 SRR1542717_clean.fq.gz</span><br><span class="line">68M Apr 29 16:30 SRR1542718_clean.fq.gz</span><br><span class="line">30M Apr 29 16:30 SRR1542719_clean.fq.gz</span><br></pre></td></tr></table></figure>
<p>前面的步骤，需要自行研究里面的细节哦。</p>
<h3 id="比对-MIRBASE数据库下载的序列-定量"><a href="#比对-MIRBASE数据库下载的序列-定量" class="headerlink" title="比对 MIRBASE数据库下载的序列 +定量"></a>比对 MIRBASE数据库下载的序列 +定量</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">mature=/home/yb77613/reference/miRNA/hsa-mature-bowtie-index</span><br><span class="line">hairpin=/home/yb77613/reference/miRNA/hsa-hairpin-bowtie-index</span><br><span class="line">ls *_clean.fq.gz |<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$id</span> </span><br><span class="line">bowtie -n 0 -m1 --best --strata <span class="variable">$mature</span> <span class="variable">$id</span> -S <span class="variable">$&#123;id&#125;</span>_matrue.sam </span><br><span class="line">bowtie -n 0 -m1 --best --strata <span class="variable">$hairpin</span> <span class="variable">$id</span> -S <span class="variable">$&#123;id&#125;</span>_hairpin.sam </span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">ls *.sam|<span class="keyword">while</span> <span class="built_in">read</span> id ;<span class="keyword">do</span> (samtools sort -O bam -@ 5 -o $(basename <span class="variable">$&#123;id&#125;</span> <span class="string">&quot;.sam&quot;</span>).bam <span class="variable">$&#123;id&#125;</span>);<span class="keyword">done</span></span><br><span class="line">rm *.sam</span><br></pre></td></tr></table></figure>
<p>使用miRNA序列比对的<strong>推荐参数</strong>： -n 0 -m1 —best —strata ，理由参考：</p>
<p>对其中一个样本的比对的log日志如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">SRR1542714_clean.fq.gz</span><br><span class="line"># reads processed: 1520320</span><br><span class="line"># reads with at least one reported alignment: 331645 (21.81%)</span><br><span class="line"># reads that failed to align: 1145386 (75.34%)</span><br><span class="line"># reads with alignments suppressed due to -m: 43289 (2.85%)</span><br><span class="line">Reported 331645 alignments</span><br><span class="line"># reads processed: 1520320</span><br><span class="line"># reads with at least one reported alignment: 331482 (21.80%)</span><br><span class="line"># reads that failed to align: 1008271 (66.32%)</span><br><span class="line"># reads with alignments suppressed due to -m: 180567 (11.88%)</span><br><span class="line">Reported 331482 alignments</span><br></pre></td></tr></table></figure>
<p>可以看到，比对率还是蛮低的，但是可以调整参数（允许错配碱基）的数量。</p>
<p>得到的bam文件如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">29M Apr 29 20:15 SRR1542714_clean.fq.gz_hairpin.bam</span><br><span class="line">28M Apr 29 20:15 SRR1542714_clean.fq.gz_matrue.bam</span><br><span class="line">27M Apr 29 20:15 SRR1542715_clean.fq.gz_hairpin.bam</span><br><span class="line">26M Apr 29 20:15 SRR1542715_clean.fq.gz_matrue.bam</span><br><span class="line">36M Apr 29 20:15 SRR1542716_clean.fq.gz_hairpin.bam</span><br><span class="line">35M Apr 29 20:15 SRR1542716_clean.fq.gz_matrue.bam</span><br><span class="line">13M Apr 29 20:15 SRR1542717_clean.fq.gz_hairpin.bam</span><br><span class="line">13M Apr 29 20:15 SRR1542717_clean.fq.gz_matrue.bam</span><br><span class="line">49M Apr 29 20:15 SRR1542718_clean.fq.gz_hairpin.bam</span><br><span class="line">48M Apr 29 20:15 SRR1542718_clean.fq.gz_matrue.bam</span><br><span class="line">22M Apr 29 20:15 SRR1542719_clean.fq.gz_hairpin.bam</span><br><span class="line">22M Apr 29 20:15 SRR1542719_clean.fq.gz_matrue.bam</span><br></pre></td></tr></table></figure>
<p>批量走 samtools idxstats 得到表达量矩阵：</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam |xargs -i samtools index &#123;&#125;</span><br><span class="line">ls *.bam|<span class="keyword">while</span> <span class="built_in">read</span> id ;<span class="keyword">do</span> (samtools idxstats <span class="variable">$&#123;id&#125;</span> &gt; <span class="variable">$&#123;id&#125;</span>.txt );<span class="keyword">done</span></span><br><span class="line"><span class="comment"># samtools view matrue.bam |cut -f 3 |sort |uniq -c</span></span><br></pre></td></tr></table></figure>
<h3 id="比对参考基因组-定量"><a href="#比对参考基因组-定量" class="headerlink" title="比对参考基因组+定量"></a>比对参考基因组+定量</h3><figure class="highlight sh"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">index=/home/yb77613/reference/index/bowtie/hg38 </span><br><span class="line">ls *_clean.fq.gz |<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$id</span> </span><br><span class="line">bowtie2 -p 4 -x <span class="variable">$index</span> -U <span class="variable">$id</span> | samtools sort -@ 4 -o <span class="variable">$&#123;id&#125;</span>_genome.bam -</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p>使用默认参数，对其中一个样本的比对的log日志如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">SRR1542715_clean.fq.gz </span><br><span class="line">1520320 reads; of these:</span><br><span class="line"> 1520320 (100.00%) were unpaired; of these:</span><br><span class="line"> 123178 (8.10%) aligned 0 times</span><br><span class="line"> 333700 (21.95%) aligned exactly 1 time</span><br><span class="line"> 1063442 (69.95%) aligned &gt;1 times</span><br><span class="line">91.90% overall alignment rate</span><br></pre></td></tr></table></figure>
<p>可以看到，这个时候的比对率不得了，得到的bam文件如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">34M Apr 29 20:30 SRR1542714_clean.fq.gz_genome.bam</span><br><span class="line">31M Apr 29 20:30 SRR1542715_clean.fq.gz_genome.bam</span><br><span class="line">42M Apr 29 20:31 SRR1542716_clean.fq.gz_genome.bam</span><br><span class="line">15M Apr 29 20:31 SRR1542717_clean.fq.gz_genome.bam</span><br><span class="line">57M Apr 29 20:32 SRR1542718_clean.fq.gz_genome.bam</span><br><span class="line">25M Apr 29 20:32 SRR1542719_clean.fq.gz_genome.bam</span><br></pre></td></tr></table></figure>
<p>然后定量，需要使用下面的命令和参数，都是需要看软件文档摸索的。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">gtf&#x3D;&#x2F;home&#x2F;yb77613&#x2F;reference&#x2F;miRNA&#x2F;hsa.gff3</span><br><span class="line">bin_featureCounts&#x3D;&quot;&#x2F;home&#x2F;yb77613&#x2F;biosoft&#x2F;featureCounts&#x2F;subread-1.5.3-Linux-x86_64&#x2F;bin&#x2F;featureCounts&quot;;</span><br><span class="line"></span><br><span class="line">$bin_featureCounts -T 4 -F gff -M -t miRNA -g Name -a $gtf -o all.counts.mature.txt *genome* 1&gt;counts.mature.log 2&gt;&amp;1</span><br><span class="line"></span><br><span class="line">$bin_featureCounts -T 4 -F gff -M -t miRNA_primary_transcript -g Name -a $gtf -o all.counts.hairpin.txt *genome* 1&gt;counts.hairpin.log 2&gt;&amp;1</span><br></pre></td></tr></table></figure>
<p>这样我们就有不同的定量流程拿到的不同的表达矩阵啦。</p>
<h3 id="比较两个定量方式的表达矩阵差异"><a href="#比较两个定量方式的表达矩阵差异" class="headerlink" title="比较两个定量方式的表达矩阵差异"></a>比较两个定量方式的表达矩阵差异</h3><p>在featureCounts软件的结果文件 all.counts.id.txt 里面的信息如下：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">hsa-miR-12136 chr1 632668 632685 - 18 26 21 50 6 78 19</span><br><span class="line">hsa-miR-34a-5p chr1 9151735 9151756 - 22 3009 9494 3467 4136 5299 7097</span><br><span class="line">hsa-miR-34a-3p chr1 9151693 9151714 - 22 88 153 100 86 126 132</span><br></pre></td></tr></table></figure>
<p>查询其中一个，发现featureCounts软件定量少了80%，比如hsa-miR-12136在featureCounts流程，是 18 26 21 50 6 78 19，但是在miRBase数据库流程，都是五倍以上的表达量。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">SRR1542714_clean.fq.gz_matrue.bam.txt:hsa-miR-12136 18 104 0</span><br><span class="line">SRR1542715_clean.fq.gz_matrue.bam.txt:hsa-miR-12136 18 182 0</span><br><span class="line">SRR1542716_clean.fq.gz_matrue.bam.txt:hsa-miR-12136 18 109 0</span><br><span class="line">SRR1542717_clean.fq.gz_matrue.bam.txt:hsa-miR-12136 18 44 0</span><br><span class="line">SRR1542718_clean.fq.gz_matrue.bam.txt:hsa-miR-12136 18 235 0</span><br><span class="line">SRR1542719_clean.fq.gz_matrue.bam.txt:hsa-miR-12136 18 92 0</span><br></pre></td></tr></table></figure>
<p>查询另外2个，有意思的是其中一个发现featureCounts软件定量多了1倍，另外一个多了2倍。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">SRR1542714_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-5p 22 1745 0</span><br><span class="line">SRR1542715_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-5p 22 5825 0</span><br><span class="line">SRR1542716_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-5p 22 2015 0</span><br><span class="line">SRR1542717_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-5p 22 2605 0</span><br><span class="line">SRR1542718_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-5p 22 3081 0</span><br><span class="line">SRR1542719_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-5p 22 4413 0</span><br><span class="line"></span><br><span class="line">SRR1542714_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-3p 22 25 0</span><br><span class="line">SRR1542715_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-3p 22 69 0</span><br><span class="line">SRR1542716_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-3p 22 21 0</span><br><span class="line">SRR1542717_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-3p 22 27 0</span><br><span class="line">SRR1542718_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-3p 22 37 0</span><br><span class="line">SRR1542719_clean.fq.gz_matrue.bam.txt:hsa-miR-34a-3p 22 57 0</span><br></pre></td></tr></table></figure>
<p>比如，hsa-miR-34a-5p在featureCounts流程，是3009 9494 3467 4136 5299 7097，但是在miRBase数据库流程，都只有一半的表达量。<br>对hsa-miR-34a-3p来说，在featureCounts流程，是88 153 100 86 126 132，但是在miRBase数据库流程，都是两倍的表达量。<br>这样的定量差异，理论上是不会改变该miRNA的差异分析结果，因为是同样的<strong>膨胀系数</strong>。<br>但是，这样的不确定性，让我们的数据分析，蒙上了一层阴影。</p>
<h1 id="文献选择与解读"><a href="#文献选择与解读" class="headerlink" title="文献选择与解读"></a>文献选择与解读</h1><p>前些天逛bioStar论坛的时候看到了一个问题，是关于miRNA分析，提问者从NCBI的SRA数据下载文献提供的原始数据，然后处理的时候有些不懂，我看到他列出的数据是iron torrent测序仪的，而且我以前还没玩过miRNA-seq的数据分析， 就抽空自学了一下。因为我有RNA-seq的基础，所以理解学习起来比较简单。特记录一下自己的学习过程，希望对后学者有帮助。</p>
<p>这里选择的文章是2014年发表的，作者用ET-1刺激human iPSCs (hiPSC-CMs) 细胞前后，想看看 miRNA和mRNA表达量的变化，我并没有细看该文章的生物学意义，仅仅从数据分析的角度解读一下这篇文章，mRNA表达量用的是Affymetrix Human Genome U133 Plus 2.0 Array，分析起来特别容易，就是得到表达矩阵，然后用limma这个包找找差异表达基因即可。但是mRNA分析起来就有点麻烦了，作者用的是iron torrent测序仪，但是从SRA数据中心下载的是已经去掉接头的测序数据，fastq格式的，所以这里其实并不需要考虑测序仪的特异性。</p>
<p>关于该文章的几个资料收集如下：</p>
<blockquote>
<p>## paper : <a target="_blank" rel="noopener" href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0108051">http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0108051</a><br>## Aggarwal P, Turner A, Matter A, Kattman SJ et al. RNA expression profiling of human iPSC-derived cardiomyocytes in a cardiac hypertrophy model. PLoS One 2014;9(9):e108051. PMID: 25255322<br>## The accession numbers are 1. SuperSeries (mRNA+miRNA) - GSE60293<br>## 2. mRNA expression array - GSE60291  (Affymetrix Human Genome U133 Plus 2.0 Array)<br>## 3. miRNA-Seq - GSE60292  (Ion Torrent)<br>## GEO  : <a target="_blank" rel="noopener" href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60292">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60292</a><br>## FTP  : <a href="ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP045/SRP045420">ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP045/SRP045420</a></p>
</blockquote>
<p>仔细看看该文章做了哪些分析，然后才能自己模仿，得到同样的数据分析结果。</p>
<p>该文章处理数据的流程是：<br>Ion Torrent’s Torrent Suite version 3.6 was used for basecalling<br>Raw sequencing reads were aligned using the <strong>SHRiMP2 aligner</strong> and were aligned against the human reference genome <strong>(hg19)</strong> for novel miRNA prediction and then against a custom reference sequence file containing <strong>miRBase v.20 known human miRNA hairpins, tRNA, rRNA,</strong> adapter sequences and predicted novel miRNA sequences.(Genome_build: <strong>hg19, miRBase v.20 human miRNA hairpins</strong>)<br>The <strong>miRDeep2 package (default parameters)</strong> was used to predict novel (as yet undescribed) miRNAs<br>Alignments with less than 17 bp matches and a custom 3′ end phred q-score threshold of 17 were filtered out.<br>miRNA quanitification was done using <strong>HTSeq v0.5.3p3</strong> using the default union parameter.<br>Differential miRNA expression was analyzed using the <strong>DESeq (v.1.12.1) R/Bioconductor package</strong><br>In this study, differentially expressed genes that had a false discovery rate cutoff at 10% (FDR&lt; = 0.1), a log2 fold change greater than 1.5 and less than −1.5 were considered significant.<br>Target gene prediction was performed using the <strong>TargetScan (version 6.2)</strong> database<br>We also used <strong>miRTarBase (version 4.3),</strong> to identify targets that have been experimentally validated<br>## miR-Deep2 and miReap ## predict exact precursor sequence according from mature sequence </p>
<p>文章提到了fastq数据质量控制标准，数据比对工具，比对的参考基因组（两条比对线路），miRNA表达量的得到，新的miRNA预测，miRNA靶基因预测，这也是我们学习miRNA-seq的数据分析的标准套路， 而且作者给出了所有的分析结果，我们完全可以通过自己的学习来重现他的分析过程。</p>
<p>Supplementary_files_format_and_content: <strong>tab-delimited text files containing raw read counts for known mature human miRNAs.（表达矩阵）</strong><br>We detected <strong>836 known human mature miRNAs</strong> in the control-CMs and <strong>769 in the ET1-CMs</strong><br>Based on our miRNA-Seq data, we predicted <strong>506 sequences to be potentially novel, as yet undescribed miRNAs.</strong><br>In order to validate the expression profiles of the miRNAs detected, <strong>we performed RT-qPCR on a subset of five known human mature and five of our predicted novel miRNAs.</strong><br>we obtained a total of <strong>1,922 predicted miRNA-mRNA pairs</strong> represented by 309 genes and 174 known mature human miRNAs.  （）</p>
<p>当然仅仅是套路分析无法发文章的，所以他结合了 miRNA和mRNA 进行网络分析，还做了少量湿实验来验证，最后还扯了一些生物学意义，当然这种纯粹理论分析肯定不好扯什么治病救人的伟大理想。</p>
<h1 id="学习资料的搜集"><a href="#学习资料的搜集" class="headerlink" title="学习资料的搜集"></a>学习资料的搜集</h1><p>因为我也是完全从零开始入门miRNA-seq分析，所以收集的资料比较齐全，我首先看了部分中文资料，了解了miRNA测序是怎么回事，该分析什么，然后主要围绕着上一篇提到的文献里面的分析步骤来搜索资料。传送门：<a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/1693.html">自学miRNA-seq分析第一讲~文献选择与解</a></p>
<p>我首先拿到了miRNA定义：<a target="_blank" rel="noopener" href="http://nar.oxfordjournals.org/content/34/suppl_1/D135.full">http://nar.oxfordjournals.org/content/34/suppl_1/D135.full </a>，当然基本上每个研究miRNA的文章都会在前言里面写到这个，我只是随意列出一个而已。</p>
<p>MicroRNAs (miRNAs) are <strong>small RNA molecules</strong>, which are <strong>∼22 nt sequences</strong> that have an important role in the translational regulation and degradation of mRNA by the base’s pairing to the 3′-untranslated regions (3′-UTR) of the mRNAs. The miRNAs are derived from the <strong>precursor transcripts of ∼70–120 nt sequences</strong>, which fold to <strong>form as stem–loop structures</strong>, which are thought to be highly conserved in the evolution of genomes. Previous analyses have suggested that <strong>∼1% of all human genes are miRNA genes,</strong> which regulate the production of protein for 10% or more of all human coding genes。</p>
<p>然后我比较纠结的问题是参考序列如何选择，因为miRNA序列很少，把它map到3G大小的人类基因组有点浪费计算资源，正好我的服务器又坏了，不想太麻烦，想用自己的个人电脑搞定这个学习过程。我看到很多帖子提到的都是比对到参考miRNA数据库(miRNA count: 28645 entries)，用bowtie ：  <a target="_blank" rel="noopener" href="http://www.mirbase.org/">http://www.mirbase.org/</a> ，从这个数据库，我明白了前体miRNA和成熟的miRNA的区别，前体miRNA长度一般是<strong>∼70–120</strong> 碱基，前体miRNA一般是茎环结果，也就是发夹结构，所以叫做hairpin。成熟之后，一般<strong>∼22 个碱基，</strong>在miRNA数据库很容易下载到这些数据，现在的miRNA版本来说，人类这个物种已知的成熟miRNA共有2588条序列，而前体miRNA共有1881条序列，我下载（下载时间2016年6月 ）的代码是：</p>
<blockquote>
<p> wget <a href="ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.gz">ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.gz</a>  <strong>##　28645　reads</strong><br> wget <a href="ftp://mirbase.org/pub/mirbase/CURRENT/mature.fa.zip">ftp://mirbase.org/pub/mirbase/CURRENT/mature.fa.zip</a>  <strong>##  35828 reads</strong><br> wget <a href="ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.zip">ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.zip</a><br> wget <a href="ftp://mirbase.org/pub/mirbase/CURRENT/genomes/hsa.gff3">ftp://mirbase.org/pub/mirbase/CURRENT/genomes/hsa.gff3</a> ##<br> wget <a href="ftp://mirbase.org/pub/mirbase/CURRENT/miFam.dat.zip">ftp://mirbase.org/pub/mirbase/CURRENT/miFam.dat.zip</a></p>
<p> grep sapiens mature.fa |wc  <strong># 2588</strong><br> grep sapiens hairpin.fa |wc    <strong># 1881</strong><br>## Homo sapiens<br>perl -alne ‘{if(/^&gt;/){if(/Homo/){$tmp=1}else{$tmp=0}};next if $tmp!=1;<strong>s/U/T/g</strong> if !/&gt;/;print }’ hairpin.fa &gt;hairpin.human.fa<br>perl -alne ‘{if(/^&gt;/){if(/Homo/){$tmp=1}else{$tmp=0}};next if $tmp!=1;<strong>s/U/T/g</strong> if !/&gt;/;print }’ mature.fa &gt;mature.human.fa<br>这里值得一提的是miRBase数据库下载的序列，居然都是用U表示的，也就是说就是miRNA序列，而不是转录成该miRNA的基因序列，而我们测序的都是基因序列。</p>
</blockquote>
<p>通过这个代码制作的 hairpin.human.fa 和 mature.human.fa 就是本次数据分析的参考基因组。<br>搜集资料的过程中，我看到了一篇文献讲挖掘1000genomes的数据找到位于miRNA的snp位点，<a target="_blank" rel="noopener" href="https://genomemedicine.biomedcentral.com/articles/10.1186/gm363">https://genomemedicine.biomedcentral.com/articles/10.1186/gm363</a> ，看起来比较新奇，不过跟本次学习过程没有关系，我就是记录一下，有空回来学习学习。<br>同时，我看到了一些博客讲解如何分析miRNA数据：<a target="_blank" rel="noopener" href="http://genomespot.blogspot.com/2013/08/quick-alignment-of-microrna-seq-data-to.html">http://genomespot.blogspot.com/2013/08/quick-alignment-of-microrna-seq-data-to.html</a><br>还有很多公司讲数据分析流程：</p>
<blockquote>
<p><a target="_blank" rel="noopener" href="http://bioinfo5.ugr.es/miRanalyzer/miRanalyzer_tutorial.html">http://bioinfo5.ugr.es/miRanalyzer/miRanalyzer_tutorial.html</a><br><a target="_blank" rel="noopener" href="http://www.partek.com/sites/default/files/Assets/UserGuideMicroRNAPipeline.pdf">http://www.partek.com/sites/default/files/Assets/UserGuideMicroRNAPipeline.pdf</a><br><a target="_blank" rel="noopener" href="http://partek.com/Tutorials/microarray/microRNA/miRNA_tutorial.pdf">http://partek.com/Tutorials/microarray/microRNA/miRNA_tutorial.pdf</a><br><a target="_blank" rel="noopener" href="http://www.arraystar.com/reviews/microrna-sequencing-data-analysis-guideline/">http://www.arraystar.com/reviews/microrna-sequencing-data-analysis-guideline/</a><br><a target="_blank" rel="noopener" href="http://bioinfo5.ugr.es/sRNAbench/sRNAbench_tutorial.pdf">http://bioinfo5.ugr.es/sRNAbench/sRNAbench_tutorial.pdf</a><br><a target="_blank" rel="noopener" href="http://seqcluster.readthedocs.io/mirna_annotation.html">http://seqcluster.readthedocs.io/mirna_annotation.html</a></p>
</blockquote>
<p>耶鲁大学好像做得不错： <a target="_blank" rel="noopener" href="http://www.yale.edu/giraldezlab/miRNA.html">http://www.yale.edu/giraldezlab/miRNA.html</a><br>中国有个南方基因： <a target="_blank" rel="noopener" href="http://www.southgene.com/newsshow.php?cid=55&amp;id=73">http://www.southgene.com/newsshow.php?cid=55&amp;id=73</a><br>miRNA研究整套方案 <a target="_blank" rel="noopener" href="http://wenku.baidu.com/view/5f38577a31b765ce05081429.html?re=view">http://wenku.baidu.com/view/5f38577a31b765ce05081429.html?re=view</a><br>Biostar 讨论帖子：<br><a target="_blank" rel="noopener" href="https://www.biostars.org/p/3344/">https://www.biostars.org/p/3344/</a><br><a target="_blank" rel="noopener" href="https://www.biostars.org/p/98486/">https://www.biostars.org/p/98486/</a><br>miRNA-seq数据处理实战指南：　<a target="_blank" rel="noopener" href="http://bib.oxfordjournals.org/content/early/2015/04/17/bib.bbv019.full">http://bib.oxfordjournals.org/content/early/2015/04/17/bib.bbv019.full</a><br>直接用一个包也可以搞定：　<a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/bioc/html/easyRNASeq.html">http://bioconductor.org/packages/release/bioc/html/easyRNASeq.html</a><br>ｇｉｔｈｕｂ流程：miRNA Analysis Pipeline v0.2.7　　　<a target="_blank" rel="noopener" href="https://github.com/bcgsc/mirna/tree/master/v0.2.7">https://github.com/bcgsc/mirna/tree/master/v0.2.7</a><br><a target="_blank" rel="noopener" href="https://tools.thermofisher.com/content/sfs/manuals/CO25176_0512.pdf">https://tools.thermofisher.com/content/sfs/manuals/CO25176_0512.pdf</a><br>miRNA annotation　　：　　<a target="_blank" rel="noopener" href="http://seqcluster.readthedocs.io/mirna_annotation.html">http://seqcluster.readthedocs.io/mirna_annotation.html</a><br>开发的网页版分析工具：　<a target="_blank" rel="noopener" href="https://wiki.uio.no/projects/clsi/images/2/2f/HTS_2014_miRNA_analysis_Lifeportal_14_final.pdf">https://wiki.uio.no/projects/clsi/images/2/2f/HTS_2014_miRNA_analysis_Lifeportal_14_final.pdf</a><br>Ｒ　ｐａｃｋａｇｅ　也很好用：　<a target="_blank" rel="noopener" href="http://bioinf.wehi.edu.au/subread-package/SubreadUsersGuide.pdf">http://bioinf.wehi.edu.au/subread-package/SubreadUsersGuide.pdf</a><br>一个培训：　<a target="_blank" rel="noopener" href="http://www.training.prace-ri.eu/uploads/tx_pracetmo/NGSdataAnalysisWithChipster.pdf">http://www.training.prace-ri.eu/uploads/tx_pracetmo/NGSdataAnalysisWithChipster.pdf</a><br>可视化IGV User Guide：　　<a target="_blank" rel="noopener" href="http://www.broadinstitute.org/igv/book/export/html/6">http://www.broadinstitute.org/igv/book/export/html/6</a><br>比较特殊的是新的miRNA预测，miRNA靶基因预测，这块研究太多软件了，并没有成型的流程和标准。</p>
<h1 id="公共测序数据下载"><a href="#公共测序数据下载" class="headerlink" title="公共测序数据下载"></a>公共测序数据下载</h1><p>前面已经讲到了该文章的数据已经上传到NCBI的SRA数据中心，所以直接根据索引号下载，然后用SRAtoolkit转出我们想要的fastq测序数据即可。下载的数据一般要进行质量控制，可视化展现一下质量如何，然后根据大题测序质量进行简单过滤。所以需要提前安装一些软件来完成这些任务，包括： sratoolkit /fastx_toolkit /fastqc/bowtie2/hg19/miRBase/SHRiMP</p>
<p>下面是我用新服务器下载安装软件的一些代码记录，因为fastx_toolkit /fastqc我已经安装过，就不列代码了，还有miRBase的下载，我在前面第二讲里面提到过，传送门：<a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/1697.html">自学miRNA-seq分析第二讲~学习资料的搜集</a></p>
<blockquote>
<p>## pre-step: download sratoolkit /fastx_toolkit_0.0.13/fastqc/bowtie2/hg19/miRBase/SHRiMP<br>## <a target="_blank" rel="noopener" href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software">http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software</a><br>## <a target="_blank" rel="noopener" href="http://www.ncbi.nlm.nih.gov/books/NBK158900/">http://www.ncbi.nlm.nih.gov/books/NBK158900/</a><br> ## 我这里特意挑选的二进制版本程序下载的，这样直接解压就可以用，但是需要挑选适合自己的操作系统的程序。<br>cd ~/biosoft<br>mkdir sratoolkit &amp;&amp;  cd sratoolkit<br>wget <a target="_blank" rel="noopener" href="http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.6.3/sratoolkit.2.6.3-centos_linux64.tar.gz">http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.6.3/sratoolkit.2.6.3-centos_linux64.tar.gz</a><br>##<br>##  Length: 63453761 (61M) [application/x-gzip]<br>##  Saving to: “sratoolkit.2.6.3-centos_linux64.tar.gz”<br>tar zxvf <strong>sratoolkit.2.6.3-centos_linux64.tar.gz</strong><br>cd ~/biosoft<br>mkdir bowtie &amp;&amp;  cd bowtie<br>wget <a target="_blank" rel="noopener" href="https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.2.9/bowtie2-2.2.9-linux-x86_64.zip/download">https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.2.9/bowtie2-2.2.9-linux-x86_64.zip/download</a></p>
<p>#Length: 27073243 (26M) [application/octet-stream]<br>#Saving to: “download”<br> mv download  bowtie2-2.2.9-linux-x86_64.zip<br> unzip <strong>bowtie2-2.2.9-linux-x86_64.zip</strong><br>## <a target="_blank" rel="noopener" href="http://compbio.cs.toronto.edu/shrimp/">http://compbio.cs.toronto.edu/shrimp/</a><br>mkdir SHRiMP &amp;&amp;  cd SHRiMP<br>wget <a target="_blank" rel="noopener" href="http://compbio.cs.toronto.edu/shrimp/releases/SHRiMP_2_2_3.lx26.x86_64.tar.gz">http://compbio.cs.toronto.edu/shrimp/releases/SHRiMP_2_2_3.lx26.x86_64.tar.gz</a><br>tar zxvf <strong>SHRiMP_2_2_3.lx26.x86_64.tar.gz</strong><br>cd SHRiMP_2_2_3<br>export SHRIMP_FOLDER=$PWD  ## 这个软件使用的时候比较奇葩，需要设置到环境变量，不能简单的调用全路径</p>
</blockquote>
<p>SHRiMP这个软件比较小众，我也是第一次听说过，本来我计划是能用bowtie搞定，就不麻烦了，但是第一次比对出了一个bug，就是下载的miRNA序列里面的U没有转换成T，所以导致比对率非常之低，所以我不得不根据文章里面记录的软件SHRiMP 来做比对，最后发现比对率完全没有改善，搞得我都在怀疑是不是作者乱来了。</p>
<p>下面是下载数据，质量控制的代码，希望大家可以照着运行一下：</p>
<blockquote>
<p>## step1 : download raw data<br>mkdir miRNA<em>test &amp;&amp; cd miRNA_test<br>echo {14..19} |sed ‘s/ /\n/g’ |while read id; \<br>do  wget “<a href="ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP045/SRP045420/SRR15427$id/SRR15427$id.sra">ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP045/SRP045420/SRR15427$id/SRR15427$id.sra</a>“  ;\<br>done<br>## step2 :  change sra data to fastq files.<br>## 主要是用shell脚本来批量下载<br>ls <em>sra |while read id; do ~/biosoft/sratoolkit/sratoolkit.2.6.3-centos_linux64/bin/fastq-dump $id;done<br>rm </em>sra<br>##  33M —&gt; 247M<br>#Read 1866654 spots for SRR1542714.sra<br>#Written 1866654 spots for SRR1542714.sra<br>## step3 : download the results from paper<br>## <a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/1571.html">http://www.bio-info-trainee.com/1571.html</a><br>## <a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE1nnn/GSE1009/suppl/GSE1009_RAW.tar">ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE1nnn/GSE1009/suppl/GSE1009_RAW.tar</a><br>mkdir paper_results &amp;&amp; cd paper_results<br>wget <a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE60nnn/GSE60292/suppl/GSE60292_RAW.tar">ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE60nnn/GSE60292/suppl/GSE60292_RAW.tar</a><br>## tar xvf GSE60292_RAW.tar<br>ls *gz |while read id ; do (echo $id;zcat $id | cut -f 2 |perl -alne ‘{$t+=$</em>;}END{print $t}’);done<br>ls <em>gz |xargs gunzip<br>## step4 : quality assessment<br>ls </em>fastq | while read id ; do ~/biosoft/fastqc/FastQC/fastqc $id;done<br>## Sequence length 8-109<br>## %GC 52<br>## Adapter Content passed<br>## write a script : :: cat &gt;filter.sh<br>ls <em>fastq |while read id<br>do<br>echo $id<br>~/biosoft/fastx_toolkit_0.0.13/bin/fastq_quality_filter <strong>-v -q 20 -p 80 -Q33</strong> -i $id -o tmp ;<br>~/biosoft/fastx_toolkit_0.0.13/bin/fastx_trimmer <strong>-v -f 1 -l 27</strong> <strong>-i tmp  -Q33 -z</strong> -o ${id%%.</em>}_clean.fq.gz ;<br>done<br>rm tmp<br>## <strong>discarded 12%~~49%%</strong><br>ls <em>_clean.fq.gz | while read id ; do ~/biosoft/fastqc/FastQC/fastqc $id;done<br>mkdir QC_results<br>mv </em>zip *html QC_results</p>
</blockquote>
<p>这个代码是我自己根据文章的理解写出的，因为我本身不擅长miRNA数据分析，所以在进行QC的时候参数选择可能并不是那么友好，如果有高手能指正就最好了，可以直接打我电话告诉我，或者发邮箱给我，邮箱用户名是jmzeng1314，是163邮箱。</p>
<p>~/biosoft/fastx_toolkit_0.0.13/bin/fastq_quality_filter <strong>-v -q 20 -p 80 -Q33</strong> -i $id -o tmp ;<br>~/biosoft/fastx_toolkit_0.0.13/bin/fastx_trimmer <strong>-v -f 1 -l 27</strong> <strong>-i tmp  -Q33 -z</strong> -o ${id%%.*}_clean.fq.gz ;<br>最后得到的clean.fq.gz系列文件，就是我需要进行比对的序列啦。</p>
<h1 id="测序数据比对"><a href="#测序数据比对" class="headerlink" title="测序数据比对"></a>测序数据比对</h1><p>序列比对是大多数类型数据分析的核心，如果要利用好测序数据，比对细节非常重要，我这里只是研读一篇文章也就没有对比对细节过多考虑，只是列出自己的代码和自己的几点思考，力求重现文章作者的分析结果。对miRNA-seq数据有两条比对策略，一种是下载miRBase数据库里面的已知miRNA序列来进行比对，一种直接比对到参考基因组(比如人类的是hg19/hg38)，前面的比对非常简单，而且很容易就可以数出已经的所以miRNA序列的表达量，后面的比对有点耗时，而且算表达量的时候也不是很方便，但是它有个有点是可以来预测新的miRNA，所以大多数文章都会把这两条路给走一下。</p>
<p>本文选择的是SHRiMP这个小众软件，起初我并没有在意，就用的bowtie2而已，参考基因组我这里因为服务器原因，就用了miRBase数据库下载的人类的参考序列，现在的miRNA版本来说，人类这个物种已知的成熟miRNA共有2588条序列，而前体miRNA共有1881条序列，我下载（下载时间2016年6月 ）的代码见<a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/1697.html"> 自学miRNA-seq分析第二讲~学习资料的搜集</a> ，下面比对所用到的软件已经序列在我的： <a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/1703.html">自学miRNA-seq分析第三讲~公共测序数据下载</a></p>
<blockquote>
<p>## step5 : alignment to miRBase v21 (hairpin.human.fa/mature.human.fa )<br>#### step5.1 using bowtie2 to do alignment<br>mkdir  bowtie2_index &amp;&amp;  cd bowtie2_index<br>~/biosoft/bowtie/bowtie2-2.2.9/bowtie2-build ../hairpin.human.fa hairpin_human<br>~/biosoft/bowtie/bowtie2-2.2.9/bowtie2-build ../mature.human.fa  mature_human<br>ls <em>_clean.fq.gz | while read id ; do  ~/biosoft/bowtie/bowtie2-2.2.9/bowtie2 -x miRBase/bowtie2_index/hairpin_human -U $id  -S ${id%%.</em>}.hairpin.sam ; done<br><strong>## overall alignment rate:  10.20% / 5.71%/ 10.18%/ 4.36% / 10.02% / 4.95%  (before convert U to T )</strong><br><strong>## overall alignment rate:  51.77% / 70.38%/51.45% /61.14%/ 52.20% / 65.85% (after convert U to T )</strong><br>ls <em>_clean.fq.gz | while read id ; do  ~/biosoft/bowtie/bowtie2-2.2.9/bowtie2 -x miRBase/bowtie2_index/mature_human  -U $id  -S ${id%%.</em>}.mature.sam ; done<br><strong>## overall alignment rate:  6.67% / 3.78% / 6.70% / 2.80%/ 6.55% / 3.23%   (before convert U to T )</strong><br><strong>## overall alignment rate:  34.94% / 46.16%/ 35.00%/ 38.50% / 35.46% /42.41%(after convert U to T )</strong><br>#### step5.2 using SHRiMP to do alignment<br>##   <a target="_blank" rel="noopener" href="http://compbio.cs.toronto.edu/shrimp/README">http://compbio.cs.toronto.edu/shrimp/README</a><br>##   3.5 Mapping cDNA reads against a miRNA database<br>cd ~/biosoft/SHRiMP/SHRiMP_2_2_3<br>export SHRIMP_FOLDER=$PWD<br>cd -<br>##　　We project the database with:<br>$SHRIMP_FOLDER/utils/project-db.py —seed 00111111001111111100,00111111110011111100,00111111111100111100,00111111111111001100,00111111111111110000 \<br> —h-flag —shrimp-mode ls miRBase/hairpin.human.fa<br>##<br>$SHRIMP_FOLDER/bin/gmapper-ls -L  hairpin.human-ls SRR1542716.fastq  —qv-offset 33  \<br>-o 1 -H -E -a -1 -q -30 -g -30 —qv-offset 33 —strata -N 8  &gt;map.out 2&gt;map.log</p>
</blockquote>
<p>大家可以看到我们把测序reads比对到前体miRNA和成熟的miRNA结果是有略微区别的，因为一个前体miRNA可以形成多个成熟的miRNA，而并不是所有的成熟的miRNA形式都被记录在数据库，所以一般推荐我们比对到前体miRNA数据库，这样还可以预测新的成熟miRNA，也是非常有意义的。</p>
<p>而且有个非常重要的一点，就是大家可以看到我把U变成T前后比对率差异非常大，这其实是一个非常蠢的错误。我就不多说了。但是做到这一步，其实可以跟文章来做验证了，文章有提到比对率，比对的序列。</p>
<p>我也是在博客里面看到这个信息的：<br>Thank you so  much!. Yes I contacted the lab-guy and he just said that trimmed the first 4 bp and last 4bp. ( as you found)<br>So  I firstly <strong>trimmed the adapter sequences</strong>(TGGAATTCTCGGGTGCCAAGGAACTCCAGTCAC)<br>And then, <strong>trimmed the first 4bp and last 4bp</strong> from reads, which <strong>leads to the 22bp peak of read-length distribution(instead of 24bp)</strong><br>Anyhow, I tried to map with bowtie2 again.<br><strong>&gt;</strong> <strong>bowtie2 —local -N 1 -L 16</strong><br><strong>-x ../miRNA_reference/hairpin_UtoT.fa</strong><br><strong>-U first4bptrimmed_A1-SmallRNA_S1_L001_R1_001_Illuminaadpatertrim.fastq</strong><br><strong>-S f4_trimmed.sam</strong><br><strong>I also changed hairpin.fa file (U to T)</strong><br>Oh.. thank you David,<br>Finallly, I got<br>2565353 reads; of these:<br>2565353 (100.00%) were unpaired; of these:<br>479292 (18.68%) aligned 0 times<br>11959 (0.47%) aligned exactly 1 time<br>2074102 (80.85%) aligned &gt;1 times<br><strong>81.32% overall alignment rate</strong></p>
<h1 id="miRNA表达量获取"><a href="#miRNA表达量获取" class="headerlink" title="miRNA表达量获取"></a>miRNA表达量获取</h1><p>拿到比对后的sam/bam文件之后，这只能算是level2的数据，一般我们给他人share我们的结果也是直接给表达矩阵的， miRNA分析跟mRNA分析类似，但是它的表达矩阵更好获取一点。如果是mRNA，我们一般会跟基因组来比较，而基因组就那24条参考染色体，想知道具体比对到了哪个基因，需要根据基因组注释文件来写程序提取表达量信息，现在比较流行的是htseq这个软件，我前面也写过教程如何安装和使用，这里就不啰嗦了。但是对于miRNA，因为我比对的就是那1881条前体miRNA序列，所以直接分析比对的sam/bam文件就可以知道每条参考miRNA序列的表达量了。 </p>
<blockquote>
<p>## step6: counts the reads which mapping to each miRNA reference.<br>## we need to exclude unmapped as well as multiple-mapped  reads<br>## XS:i:<n> Alignment score for second-best alignment. Can be negative. Can be greater than 0 in —local mode<br>## NM:i:1  ## NM i Edit distance to the reference, including ambiguous bases but excluding clipping<br>#The following command exclude unmapped (-F 4) as well as multiple-mapped (grep -v “XS:”) reads<br>#samtools view -F 4 input.bam | grep -v “XS:” | wc -l<br>## 180466//1520320<br>##cat &gt;<a target="_blank" rel="noopener" href="http://count.hairpin.sh/">count.hairpin.sh</a><br>ls <em>hairpin.sam  | while read id<br>do<br>**samtools view  -SF 4 $id |perl -alne ‘{$h{$F[2]}++}END{print “$<em>\t$h{$</em>}” foreach sort keys %h }’  &gt; ${id%%_\</em>}.hairpin.counts<strong><br>done<br>## bash <a target="_blank" rel="noopener" href="http://count.hairpin.sh/">count.hairpin.sh</a><br>##cat &gt;<a target="_blank" rel="noopener" href="http://count.mature.sh/">count.mature.sh</a><br>ls *mature.sam  | while read id<br>do
</strong>samtools view  -SF 4 $id |perl -alne ‘{$h{$F[2]}++}END{print “$<em>\t$h{$</em>}” foreach sort keys %h }’  &gt; ${id%%_*}.mature.counts**<br>done<br>## bash <a target="_blank" rel="noopener" href="http://count.mature.sh/">count.mature.sh</a></n></p>
</blockquote>
<p>上面的代码，是我自己写的脚本来算表达量，非常简单，因为我没有考虑细节，直接想得到各个样本测序数据的表达量而已。如果是比对到了参考基因组，就要根据miRNA的gff注释文件用htseq等软件来计算表达量啦。<br>得到了表达量，就可以跟文献来做比较啦：</p>
<blockquote>
<p>### step7: compare the results with paper’s<br>GSM1470353: control-CM, experiment1; Homo sapiens; miRNA-Seq  SRR1542714<br>GSM1470354: ET1-CM, experiment1; Homo sapiens; miRNA-Seq  SRR1542715<br>GSM1470355: control-CM, experiment2; Homo sapiens; miRNA-SeqSRR1542716<br>GSM1470356: ET1-CM, experiment2; Homo sapiens; miRNA-Seq SRR1542717<br>GSM1470357: control-CM, experiment3; Homo sapiens; miRNA-Seq SRR1542718<br>GSM1470358: ET1-CM, experiment3; Homo sapiens; miRNA-Seq SRR1542719<br>### 下面我用R语言来检验一下，我得到的分析结果跟文章发表的结果的区别。<br> <strong>a=read.table(“bowtie_bam/SRR1542714.mature.counts”)</strong><br> <strong>b=read.table(“paper_results/GSM1470353_iPS_010313_Unstim_known_miRNA_counts.txt”)</strong><br> plot(log(tmp[,2]),log(tmp[,3]))<br> cor(tmp[,2],tmp[,3])<br><strong>##[1] 0.8413439</strong></p>
</blockquote>
<h1 id="miRNA表达量差异分析"><a href="#miRNA表达量差异分析" class="headerlink" title="miRNA表达量差异分析"></a>miRNA表达量差异分析</h1><p>这一讲是miRNA-seq数据分析的分水岭，前面的5讲说的是读文献下载数据比对然后计算表达量，属于常规的流程分析，一般在公司测序之后都可以拿到分析结果，或者文献也会给出下载结果。但是单纯的分析一个样本意义不大，一般来说，我们做研究都是针对于不同状态下的miRNA表达量差异分析，然后做注释，功能分析，网络分析，这才是重点，也是难点。我这里就直接拿文献处理好的miRNA表达量来展示如何做下游分析，首先就是差异分析啦：根据文献，我们可以知道样本的分类情况是:</p>
<blockquote>
<p>GSM1470353: control-CM, experiment1; Homo sapiens; miRNA-Seq  SRR1542714<br>GSM1470354: ET1-CM, experiment1; Homo sapiens; miRNA-Seq  SRR1542715<br>GSM1470355: control-CM, experiment2; Homo sapiens; miRNA-SeqSRR1542716<br>GSM1470356: ET1-CM, experiment2; Homo sapiens; miRNA-Seq SRR1542717<br>GSM1470357: control-CM, experiment3; Homo sapiens; miRNA-Seq SRR1542718<br>GSM1470358: ET1-CM, experiment3; Homo sapiens; miRNA-Seq SRR1542719<br>可以看到是6个样本的测序数据，分成两组，就是ET1刺激了CM细胞系前后对比而已！</p>
</blockquote>
<p>同时，我们也拿到了这6个样本的表达矩阵，计量单位是counts的reads数，所以我们一般会选用DESeq2，edgeR这样的常用包来做差异分析，当然，做差异分析的工具还有十几个，我这里只是拿一根最顺手的举例子，就是DESeq2<br>下面的代码有点长，因为我在bioconductor系列教程里面多次提到了DESeq2使用方法，这里就只贴出代码，反正我要说的重点就是，我们进行了差异分析，然后得到差异miRNA列表</p>
<blockquote>
<p>### step8: differential expression analysis by R package for miRNA expression patterns:<br>## 文章里面提到的结果是：<br>MicroRNA sequencing revealed over 250 known and 34 predicted novel miRNAs to be differentially expressed between ET-1 stimulated and unstimulated control hiPSC-CMs.<br>## (FDR &lt; 0.1 and 1.5 fold change)<br>rm(list=ls())<br>setwd(‘J:\miRNA<em>test\paper_results’) ##把从GEO里面下载的文献结果放在这里<br>sampleIDs=c()<br>groupList=c()<br>allFiles=list.files(pattern = ‘.txt’)<br>i=allFiles[1]<br>sampleID=strsplit(i,”</em>“)[[1]][1]<br>treat=strsplit(i,”<em>“)[[1]][4]<br>dat=read.table(i,stringsAsFactors = F)<br>colnames(dat)=c(‘miRNA’,sampleID)<br>groupList=c(groupList,treat)<br>for (i in allFiles[-1]){<br>sampleID=strsplit(i,”</em>“)[[1]][1]<br>treat=strsplit(i,”_”)[[1]][4]<br>a=read.table(i,stringsAsFactors = F)<br>colnames(a)=c(‘miRNA’,sampleID)<br>dat=merge(dat,a,by=’miRNA’)<br>groupList=c(groupList,treat)<br>}<br>### 上面的代码只是为了把6个独立的表达文件给合并成一个表达矩阵<br>## we need to filter the low expression level miRNA<br>exprSet=dat[,-1]<br>rownames(exprSet)=dat[,1]<br>suppressMessages(library(DESeq2))<br>exprSet=ceiling(exprSet)<br>(colData &lt;- data.frame(row.names=colnames(exprSet), groupList=groupList))</p>
<p>## DESeq2就是这么简单的用<br>dds &lt;- DESeqDataSetFromMatrix(countData = exprSet,<br>colData = colData,<br>design = ~ groupList)<br>dds &lt;- DESeq(dds)<br>png(“qc_dispersions.png”, 1000, 1000, pointsize=20)<br>plotDispEsts(dds, main=”Dispersion plot”)<br>dev.off()<br>res &lt;- results(dds)<br>## 画一些图，相当于做QC吧<br>png(“RAWvsNORM.png”)<br>rld &lt;- rlogTransformation(dds)<br>exprSet_new=assay(rld)<br>par(cex = 0.7)<br>n.sample=ncol(exprSet)<br>if(n.sample&gt;40) par(cex = 0.5)<br>cols &lt;- rainbow(n.sample*1.2)<br>par(mfrow=c(2,2))<br>boxplot(exprSet, col = cols,main=”expression value”,las=2)<br>boxplot(exprSet_new, col = cols,main=”expression value”,las=2)<br>hist(exprSet[,1])<br>hist(exprSet_new[,1])<br>dev.off()library(RColorBrewer)<br>(mycols &lt;- brewer.pal(8, “Dark2”)[1:length(unique(groupList))])</p>
<p># Sample distance heatmap<br>sampleDists &lt;- as.matrix(dist(t(exprSet_new)))<br>#install.packages(“gplots”,repos = “<a target="_blank" rel="noopener" href="http://cran.us.r-project.org">http://cran.us.r-project.org</a>“)<br>library(gplots)<br>png(“qc-heatmap-samples.png”, w=1000, h=1000, pointsize=20)<br>heatmap.2(as.matrix(sampleDists), key=F, trace=”none”,<br>col=colorpanel(100, “black”, “white”),<br>ColSideColors=mycols[groupList], RowSideColors=mycols[groupList],<br>margin=c(10, 10), main=”Sample Distance Matrix”)<br>dev.off()</p>
<p>png(“MA.png”)<br>DESeq2::plotMA(res, main=”DESeq2”, ylim=c(-2,2))<br>dev.off()<br>## 重点就是这里啦，得到了差异分析的结果<br>resOrdered &lt;- res[order(res$padj),]<br>resOrdered=as.data.frame(resOrdered)<br>write.csv(resOrdered,”<strong>deseq2.results.csv</strong>“,quote = F)</p>
<p>##下面也是一些图，主要是看看样本之间的差异情况<br>library(limma)<br>plotMDS(log(counts(dds, normalized=TRUE) + 1))<br>plotMDS(log(counts(dds, normalized=TRUE) + 1) - log(t( t(assays(dds)[[“mu”]]) / sizeFactors(dds) ) + 1))<br>plotMDS( assays(dds)[[“counts”]] ) ## raw count<br>plotMDS( assays(dds)[[“mu”]] ) ##- fitted values.</p>
</blockquote>
<p>最后我们得到的差异分析结果：deseq2.results.csv，就可以跟进FDR和fold change来挑选符合要求的差异miRNA啦</p>
<h1 id="miRNA样本配对mRNA表达量获取"><a href="#miRNA样本配对mRNA表达量获取" class="headerlink" title="miRNA样本配对mRNA表达量获取"></a>miRNA样本配对mRNA表达量获取</h1><p>这一讲其实算不上是自学miRNA-seq分析，本质就是affymetrix的mRNA表达芯片数据分析，而且还是最常用的那种GPL570   HG-U133_Plus_2，但是因为是跟miRNA样本配对检测的，而且后面会利用到这两个数据分析结果来做共表达网络分析等等，所以就贴出对该芯片数据的分析结果。文章里面也提到了 Messenger RNA expression analysis identified 731 probe sets with significant differential expression，作者挑选的差异分析结果的显著基因列表如下：## <a target="_blank" rel="noopener" href="http://journals.plos.org/plosone/article/asset?unique&amp;id=info:doi/10.1371/journal.pone.0108051.s002">http://journals.plos.org/plosone/article/asset?unique&amp;id=info:doi/10.1371/journal.pone.0108051.s002</a><br>## mRNA expression array - GSE60291 (Affymetrix Human Genome U133 Plus 2.0 Array)</p>
<p>hgu133plus2芯片数据太常见了，可以从GEO里面下载该study的原始测序数据，然后用affy,limma包来分析，也可以直接用GEOquery包来下载作者分析好的表达矩阵，然后直接做差异分析。我这里选择的是后者，而且我跟作者分析方法有一点区别是，我先把探针都注释好了基因，然后对每个基因只挑最大表达量的基因。而作者是直接对探针为单位的的表达矩阵进行差异分析，对分析结果里面的探针进行基因注释。我这里无法给出哪种方法好的绝对评价。代码如下：</p>
<blockquote>
<p>rm(list=ls())<br>library(GEOquery)<br>library(limma)<br>GSE60291 &lt;- getGEO(‘GSE60291’, destdir=”.”,getGPL = F)</p>
<p>#下面是表达矩阵<br><strong>exprSet</strong>=exprs(GSE60291[[1]])<br>library(“annotate”)<br>GSE60291[[1]]<br>## 下面是分组信息<br>pdata=pData(GSE60291[[1]])<br><strong>treatment</strong>=factor(unlist(lapply(pdata$title,function(x) strsplit(as.character(x),”-“)[[1]][1])))<br>#treatment=relevel(treatment,’control’)<br>## 下面做基因注释<br>platformDB=’hgu133plus2.db’<br>library(platformDB, character.only=TRUE)<br>probeset &lt;- featureNames(GSE60291[[1]])<br>#EGID &lt;- as.numeric(lookUp(probeset, platformDB, “ENTREZID”))<br>SYMBOL &lt;- lookUp(probeset, platformDB, “SYMBOL”)<br>## 下面对每个基因挑选最大表达量探针<br>a=cbind(SYMBOL,exprSet)<br>## remove the duplicated probeset<br>rmDupID &lt;-function(a=matrix(c(1,1:5,2,2:6,2,3:7),ncol=6)){<br>exprSet=a[,-1]<br>rowMeans=apply(exprSet,1,function(x) mean(as.numeric(x),na.rm=T))<br>a=a[order(rowMeans,decreasing=T),]<br>exprSet=a[!duplicated(a[,1]),]<br>#<br>exprSet=exprSet[!is.na(exprSet[,1]),]<br>rownames(exprSet)=exprSet[,1]<br>exprSet=exprSet[,-1]<br>return(exprSet)<br>}<br>exprSet=rmDupID(a)<br>rn=rownames(exprSet)<br>exprSet=apply(exprSet,2,as.numeric)<br>rownames(exprSet)=rn<br>exprSet[1:4,1:4]<br>#exprSet=log(exprSet) ## based on e<br>boxplot(exprSet,las=2)<br>## 下面用limma包来进行芯片数据差异分析<br>design=model.matrix(~ treatment)<br>fit=lmFit(exprSet,design)<br>fit=eBayes(fit)<br>#vennDiagram(decideTests(fit))<br>DEG=topTable(fit,coef=2,n=Inf,adjust=’BH’)<br>dim(DEG[abs(DEG[,1])&gt;1.2 &amp; DEG[,5]&lt;0.05,]) ## 806 genes<br>write.csv(DEG,”ET1-normal.DEG.csv”)</p>
</blockquote>
<p>得到的ET1-normal.DEG.csv 文件就是我们的差异分析结果，可以跟文章提供的差异结果做比较，是几乎一模一样的！</p>
<p>如果根据logFC 1.2 p 矫正P 值0.05来挑选，可以拿到806个基因。</p>
<h1 id="miRNA-mRNA表达相关下游分析"><a href="#miRNA-mRNA表达相关下游分析" class="headerlink" title="miRNA-mRNA表达相关下游分析"></a>miRNA-mRNA表达相关下游分析</h1><p>通过前面的分析，我们已经量化了ET1刺激前后的细胞的miRNA和mRNA表达水平，也通过成熟的统计学分析分别得到了差异miRNA和mRNA，这时候我们就需要换一个参考文献了，因为前面提到的那篇文章分析的不够细致，我这里选择了浙江大学的一篇TCGA数据挖掘分析文章<a target="_blank" rel="noopener" href="http://www.nature.com/articles/srep12995">Identifying miRNA/mRNA negative regulation pairs in colorectal cancer</a>，里面首先就是查找miRNA-mRNA基因对，因为miRNA主要还是负向调控mRNA表达，所以根据我们得到的两个表达矩阵做相关性分析，很容易得到符合统计学意义的miRNA-mRNA基因对，具体分析内容如下：</p>
<blockquote>
<p>把得到的差异miRNA的表达量画一个热图，看看它是否能显著的分类<br>用miRWalk2.0等数据库或者根据来获取这些差异miRNA的validated target genes<br>然后看看这些<strong>pairs of miRNA- target genes的表达量相关系数</strong>，选取显著正相关或者负相关的pairs<br>这些被选取的pairs of miRNA- target genes拿去做<strong>富集分析</strong><br>最后这些pairs of miRNA- target genes做<strong>PPI网络分析</strong></p>
</blockquote>
<p>首先我们看第一个热图的实现：</p>
<blockquote>
<p>resOrdered=na.omit(resOrdered)<br>DEmiRNA=resOrdered[abs(resOrdered$log2FoldChange)&gt;log2(1.5) &amp; resOrdered$padj &lt;0.01 ,]<br>write.csv(resOrdered,”deseq2.results.csv”,quote = F)<br>DEmiRNAexprSet=exprSet[rownames(DEmiRNA),]<br>write.csv(DEmiRNAexprSet,’DEmiRNAexprSet.csv’)</p>
<p>DEmiRNAexprSet=read.csv(‘<strong>DEmiRNAexprSet.csv</strong>‘,stringsAsFactors = F)<br>exprSet=as.matrix(DEmiRNAexprSet[,2:7])<br>rownames(exprSet)=rownames(DEmiRNAexprSet)<br>heatmap(exprSet)<br>gplots::heatmap.2(exprSet)<br>library(pheatmap)<br>## <a target="_blank" rel="noopener" href="http://biit.cs.ut.ee/clustvis/">http://biit.cs.ut.ee/clustvis/</a></p>
</blockquote>
<p>因为我前面保存的表达量就基于counts的，所以画热图还需要进行normalization，我这里懒得弄了，就用了一个网页版工具，自动出热图<a target="_blank" rel="noopener" href="http://biit.cs.ut.ee/clustvis/">http://biit.cs.ut.ee/clustvis/</a></p>
<p><a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/wp-content/uploads/2016/07/miRNA-heatmap.png"><img src="http://www.bio-info-trainee.com/wp-content/uploads/2016/07/miRNA-heatmap.png" alt="miRNA-heatmap"></a></p>
<p>感觉还不错，可以很清楚的看到ET1刺激前后细胞中miRNA表达量变化</p>
<p>然后就是检验我们选取的感兴趣的有显著差异的miRNA的target genes，这时候有两种方法，一个是先由数据库得到已经被检验的miRNA的target genes，另一种是根据miRNA和mRNA表达量的相关性来预测。</p>
<p>用数据库来查找MiRNA的作用基因，非常多的工具，比较常用的有<strong>TargetScan/miRTarBase</strong><br>### <a target="_blank" rel="noopener" href="http://nar.oxfordjournals.org/content/early/2015/11/19/nar.gkv1258.full">http://nar.oxfordjournals.org/content/early/2015/11/19/nar.gkv1258.full</a><br>### <a target="_blank" rel="noopener" href="http://mirtarbase.mbc.nctu.edu.tw/">http://mirtarbase.mbc.nctu.edu.tw/</a><br>### <a target="_blank" rel="noopener" href="http://mirtarbase.mbc.nctu.edu.tw/cache/download/6.1/hsa_MTI.xlsx">http://mirtarbase.mbc.nctu.edu.tw/cache/download/6.1/hsa_MTI.xlsx</a><br>### <a target="_blank" rel="noopener" href="http://www.targetscan.org/vert_71/">http://www.targetscan.org/vert_71/</a> (version 7.1 (June 2016))<br>我还看到过一个整合工具： miRecords (DIANA-microT, MicroInspector, miRanda, MirTarget2, miTarget, NBmiRTar, PicTar, PITA, RNA22, RNAhybrid and TargetScan/TargertScanS)里面提到了查找MiRNA的作用基因这一过程，高假阳性，至少被5种工具支持，才算是真的<br>还有很多类似的工具，miRWalk2，psRNATarget网页版工具，最后值得一提的是中山大学的：<a target="_blank" rel="noopener" href="http://starbase.sysu.edu.cn/panCancer.php"> starBase </a>Pan-Cancer Analysis Platform is designed for deciphering Pan-Cancer Networks of lncRNAs, miRNAs, ceRNAs and RNA-binding proteins (RBPs) by mining clinical and expression profiles of 14 cancer types (&gt;6000 samples) from The Cancer Genome Atlas (TCGA) Data Portal (all data available without limitations).虽然我没有仔细的用，但是看介绍好牛的样子，还有一个R包：miRLAB我玩了一会，它是先通过算所有配对的<strong>miRNA- genes的表达量相关系数</strong>，选取显著正相关或者负相关的pairs，然后反过来通过已知数据库来验证。</p>
<p>后面我就不讲了，主要看你得到miRNA的时候其它生物学数据是否充分，如果是癌症病人，有生存相关数据，可以做生存分析，如果你同时测了甲基化数据，可以做甲基化相关分析<del>~</del>~~~~</p>
<p>如果只是单纯的miRNA测序数据，可以回过头去研究一下de novo的miRNA预测的步骤，也是研究重点</p>
<h1 id="引用"><a href="#引用" class="headerlink" title="引用"></a>引用</h1><p><a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/tag/mirna-seq">生信菜鸟团—自学miRNA-seq</a></p>
<p>参考文章如引起任何侵权问题，可以与我<a href="mailto:liaoshuilin@outlook.com">联系</a>，谢谢。</p>

    </div>

    
    
    
	
	
	  <div>
	    <div>
  
    <div style="text-align:center;color:#bfbfbf;font-size:16px;">
      <span>-------- 本文结束 </span>
      <i class="fa fa-paw"></i>
      <span> 感谢阅读 --------</span>
    </div>
  
</div>  # 添加内容
	  </div>
	
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>本文作者： </strong>ShuilinL
  </li>
  <li class="post-copyright-link">
    <strong>本文链接：</strong>
    <a href="http://example.com/2020/12/11/2020-12-11-miRNA-seq/" title="miRNA-seq生信分析（转生信菜鸟团）">http://example.com/2020/12/11/2020-12-11-miRNA-seq/</a>
  </li>
  <li class="post-copyright-license">
    <strong>版权声明： </strong>本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> 许可协议。转载请注明出处！
  </li>
</ul>
</div>


      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/miRNA-seq/" rel="tag"># miRNA-seq</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2020/12/04/2020-12-04-p_adjust/" rel="prev" title="P值校正">
      <i class="fa fa-chevron-left"></i> P值校正
    </a></div>
      <div class="post-nav-item">
    <a href="/2020/12/28/2020-12-28-%E5%A6%82%E4%BD%95%E9%AB%98%E9%80%9F%E4%B8%8B%E8%BD%BDSRA/" rel="next" title="如何高速下载SRA">
      如何高速下载SRA <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          
    <div class="comments" id="valine-comments"></div>

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          文章目录
        </li>
        <li class="sidebar-nav-overview">
          站点概览
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#%E6%9E%84%E5%BB%BAmiRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E7%8E%AF%E5%A2%83"><span class="nav-number">1.</span> <span class="nav-text">构建miRNA-seq数据分析环境</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E9%A6%96%E5%85%88%E4%B8%8B%E8%BD%BDMIRBASE%E7%9A%84MIRNA%E5%8F%82%E8%80%83%E5%BA%8F%E5%88%97%E6%96%87%E4%BB%B6"><span class="nav-number">1.0.1.</span> <span class="nav-text">首先下载MIRBASE的MIRNA参考序列文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%84%B6%E5%90%8E%E4%BD%BF%E7%94%A8CONDA%E9%85%8D%E7%BD%AE%E5%A5%BD%E8%BD%AF%E4%BB%B6%E7%8E%AF%E5%A2%83"><span class="nav-number">1.0.2.</span> <span class="nav-text">然后使用CONDA配置好软件环境</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E9%92%88%E5%AF%B9MIRBASE%E6%95%B0%E6%8D%AE%E5%BA%93%E4%B8%8B%E8%BD%BD%E7%9A%84%E5%BA%8F%E5%88%97%E6%9E%84%E5%BB%BABOWTIE%E7%B4%A2%E5%BC%95"><span class="nav-number">1.0.3.</span> <span class="nav-text">针对MIRBASE数据库下载的序列构建BOWTIE索引</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E4%B8%8B%E8%BD%BD%E6%B5%8B%E8%AF%95%E6%95%B0%E6%8D%AE"><span class="nav-number">1.0.4.</span> <span class="nav-text">下载测试数据</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E8%B4%A8%E6%8E%A7%E5%92%8C%E6%B8%85%E6%B4%97%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE"><span class="nav-number">1.0.5.</span> <span class="nav-text">质控和清洗测序数据</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%AF%94%E5%AF%B9-MIRBASE%E6%95%B0%E6%8D%AE%E5%BA%93%E4%B8%8B%E8%BD%BD%E7%9A%84%E5%BA%8F%E5%88%97-%E5%AE%9A%E9%87%8F"><span class="nav-number">1.0.6.</span> <span class="nav-text">比对 MIRBASE数据库下载的序列 +定量</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%AF%94%E5%AF%B9%E5%8F%82%E8%80%83%E5%9F%BA%E5%9B%A0%E7%BB%84-%E5%AE%9A%E9%87%8F"><span class="nav-number">1.0.7.</span> <span class="nav-text">比对参考基因组+定量</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%AF%94%E8%BE%83%E4%B8%A4%E4%B8%AA%E5%AE%9A%E9%87%8F%E6%96%B9%E5%BC%8F%E7%9A%84%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5%E5%B7%AE%E5%BC%82"><span class="nav-number">1.0.8.</span> <span class="nav-text">比较两个定量方式的表达矩阵差异</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E6%96%87%E7%8C%AE%E9%80%89%E6%8B%A9%E4%B8%8E%E8%A7%A3%E8%AF%BB"><span class="nav-number">2.</span> <span class="nav-text">文献选择与解读</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%AD%A6%E4%B9%A0%E8%B5%84%E6%96%99%E7%9A%84%E6%90%9C%E9%9B%86"><span class="nav-number">3.</span> <span class="nav-text">学习资料的搜集</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%85%AC%E5%85%B1%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD"><span class="nav-number">4.</span> <span class="nav-text">公共测序数据下载</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E6%AF%94%E5%AF%B9"><span class="nav-number">5.</span> <span class="nav-text">测序数据比对</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#miRNA%E8%A1%A8%E8%BE%BE%E9%87%8F%E8%8E%B7%E5%8F%96"><span class="nav-number">6.</span> <span class="nav-text">miRNA表达量获取</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#miRNA%E8%A1%A8%E8%BE%BE%E9%87%8F%E5%B7%AE%E5%BC%82%E5%88%86%E6%9E%90"><span class="nav-number">7.</span> <span class="nav-text">miRNA表达量差异分析</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#miRNA%E6%A0%B7%E6%9C%AC%E9%85%8D%E5%AF%B9mRNA%E8%A1%A8%E8%BE%BE%E9%87%8F%E8%8E%B7%E5%8F%96"><span class="nav-number">8.</span> <span class="nav-text">miRNA样本配对mRNA表达量获取</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#miRNA-mRNA%E8%A1%A8%E8%BE%BE%E7%9B%B8%E5%85%B3%E4%B8%8B%E6%B8%B8%E5%88%86%E6%9E%90"><span class="nav-number">9.</span> <span class="nav-text">miRNA-mRNA表达相关下游分析</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%BC%95%E7%94%A8"><span class="nav-number">10.</span> <span class="nav-text">引用</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="ShuilinL"
      src="/images/headshot.jpg">
  <p class="site-author-name" itemprop="name">ShuilinL</p>
  <div class="site-description" itemprop="description">Bioinformatics</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">17</span>
          <span class="site-state-item-name">日志</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">17</span>
        <span class="site-state-item-name">分类</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">28</span>
        <span class="site-state-item-name">标签</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/ShuilinLiao" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;ShuilinLiao" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:liaoshuilin@outlook.com" title="E-Mail → mailto:liaoshuilin@outlook.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">ShuilinL</span>
</div>
  <div class="powered-by">由 <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Gemini</a> 强力驱动
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  
      

<script>
  if (typeof MathJax === 'undefined') {
    window.MathJax = {
      loader: {
        source: {
          '[tex]/amsCd': '[tex]/amscd',
          '[tex]/AMScd': '[tex]/amscd'
        }
      },
      tex: {
        inlineMath: {'[+]': [['$', '$']]},
        tags: 'ams'
      },
      options: {
        renderActions: {
          findScript: [10, doc => {
            document.querySelectorAll('script[type^="math/tex"]').forEach(node => {
              const display = !!node.type.match(/; *mode=display/);
              const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display);
              const text = document.createTextNode('');
              node.parentNode.replaceChild(text, node);
              math.start = {node: text, delim: '', n: 0};
              math.end = {node: text, delim: '', n: 0};
              doc.math.push(math);
            });
          }, '', false],
          insertedScript: [200, () => {
            document.querySelectorAll('mjx-container').forEach(node => {
              let target = node.parentNode;
              if (target.nodeName.toLowerCase() === 'li') {
                target.parentNode.classList.add('has-jax');
              }
            });
          }, '', false]
        }
      }
    };
    (function () {
      var script = document.createElement('script');
      script.src = '//cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js';
      script.defer = true;
      document.head.appendChild(script);
    })();
  } else {
    MathJax.startup.document.state(0);
    MathJax.texReset();
    MathJax.typeset();
  }
</script>

    

  


<script>
NexT.utils.loadComments(document.querySelector('#valine-comments'), () => {
  NexT.utils.getScript('//unpkg.com/valine/dist/Valine.min.js', () => {
    var GUEST = ['nick', 'mail', 'link'];
    var guest = 'nick,mail,link';
    guest = guest.split(',').filter(item => {
      return GUEST.includes(item);
    });
    new Valine({
      el         : '#valine-comments',
      verify     : false,
      notify     : false,
      appId      : 'MzBl7JJb4fknGe29QPTfvdou-gzGzoHsz',
      appKey     : 'SEnyER5fvAO1rUpH4njX1Vxp',
      placeholder: "输入你的评论\n昵称为必填项目，输入QQ号可以直接获取用户名和头像，并且省去了填写邮箱的麻烦！\n虽然email不是必选，但是填写了email可以收到推送通知哦！",
      avatar     : '',
      meta       : guest,
      pageSize   : '10' || 10,
      visitor    : true,
      lang       : 'zh-cn' || 'zh-cn',
      path       : location.pathname,
      recordIP   : true,
      serverURLs : ''
    });
  }, window.Valine);
});
</script>
<div class="moon-menu">
  <div class="moon-menu-items">
    
    <div class="moon-menu-item" onclick="back2bottom()">
      <i class='fa fa-chevron-down'></i>    </div>
    
    <div class="moon-menu-item" onclick="back2top()">
      <i class='fa fa-chevron-up'></i>    </div>
    
  </div>
  <div class="moon-menu-button">
    <svg class="moon-menu-bg">
      <circle class="moon-menu-cricle" cx="50%" cy="50%" r="44%"></circle>
      <circle class="moon-menu-border" cx="50%" cy="50%" r="48%"></circle>
    </svg>
    <div class="moon-menu-content">
      <div class="moon-menu-icon"><i class='fas fa-ellipsis-v'></i></div>
      <div class="moon-menu-text"></div>
    </div>
  </div>
</div><script src="/js/injector.js"></script>
</body>
</html>
